src/read.js

```diff
@@ -4,32 +4,68 @@ var copychars = require("@dcos/copychars").default;
 
 var RECORD_PATTERN = /^\d+\n.+/;
 
+function charSizeAt(string, index) {
```

Comment:
I've found this implementation that includes 4 byte chars. Do we wanna adapt the solution to use `@protobufjs/utf8`? https://github.com/dcodeIO/protobuf.js/blob/69623a91c1e4a99d5210b5295a9e5b39d9517554/lib/utf8/index.js#L15

Reply:
issue is that we dont need the length of a given string, but the string of a given length :/

Reply:
Ok, your implementation gives the same length for the Kuhn test as the `utf8`

Reply:
JavaScript uses UCS-2 and non Basic Multilingual Plane or BMP are represented in surrogate pairs. This means that `𠜎` has a length of 2 – one with a size of 3 and the other with 1.